src/main/Application.h

```diff
@@ -204,17 +204,31 @@ class Application
     // one of the following static functions, or an exhaustive switch on the
     // enum. These options give the least chance of missing or combining cases
     // incorrectly.
-    static bool modeHasOverlay(AppMode m);
-    static bool modeHasDatabase(AppMode m);
+
+    static bool modeUsesInMemoryDB(AppMode m);
```

Comment:
`modeUsesInMemoryDB` is interesting to me: do we really need it?
I *think* that what we should be doing is when the "fast meta replay" mode is enabled we:
* override the database mode to "memory" (`DATABASE=sqlite3://:memory:`)
* disable fsync
* disable historical data
* enable in memory ledger